tls/s2n_psk.c

```diff
@@ -599,3 +599,53 @@ int s2n_connection_set_psk_mode(struct s2n_connection *conn, s2n_psk_mode mode)
     conn->psk_mode_overridden = true;
     return S2N_SUCCESS;
 }
+
+int s2n_connection_get_chosen_psk(struct s2n_connection *conn, struct s2n_psk *chosen_psk)
```

Comment:
Shouldn't we also copy over the type if we want to use the get_type API on the returned chosen psk?

Reply:
A complete PSK would be the ideal return value with all its fields set and valid. Related: https://github.com/aws/s2n-tls/pull/2748#discussion_r617733424